spec/controllers/concerns/idv/document_capture_concern_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
+RSpec.describe Idv::DocumentCaptureConcern, :controller do
+  idv_document_capture_controller_class = Class.new(ApplicationController) do
+    def self.name
+      'AnonymousController'
+    end
+
+    include Idv::DocumentCaptureConcern
+
+    def show
+      render plain: 'Hello'
+    end
+  end
+
+  # def confirm_selfie_performed_if_needed
+  #   !decorated_sp_session.selfie_required? || stored_result.selfie_check_performed
+  # end
+
+  describe '#confirm_selfie_performed_if_needed' do
+    controller(idv_document_capture_controller_class) do
+    end
+
+    context 'selfie checks enabled' do
+      before do
+        decorated_sp_session = double
+        allow(decorated_sp_session).to receive(:selfie_required?).and_return(selfie_required)
+        allow(controller).to receive(:decorated_sp_session).and_return(decorated_sp_session)
+        stored_result = double
+        allow(stored_result).to receive(:selfie_check_performed).and_return(selfie_check_performed)
+        allow(controller).to receive(:stored_result).and_return(stored_result)
+      end
+
+      context 'SP requires biometric_comparison' do
+        let(:selfie_required) { true }
+
+        context 'selfie check performed' do
+          let(:selfie_check_performed) { true }
+          it 'returns true' do
+            expect(controller.confirm_selfie_performed_if_needed).to eq(true)
+          end
+        end
+
+        context 'selfie check not performed' do
+          let(:selfie_check_performed) { false }
+          it 'returns false' do
+            expect(controller.confirm_selfie_performed_if_needed).to eq(false)
+          end
+        end
+      end
+
+      context 'SP does not require biometric_comparison' do
+        let(:selfie_required) { false }
+
+        context 'selfie check performed' do
+          let(:selfie_check_performed) { true }
+          it 'returns true' do
+            expect(controller.confirm_selfie_performed_if_needed).to eq(true)
+          end
+        end
+
+        context 'selfie check not performed' do
+          let(:selfie_check_performed) { false }
+          it 'returns true' do
+            expect(controller.confirm_selfie_performed_if_needed).to eq(true)
```

Comment:
Sonia and I talked about a possible rename for this method to `selfie_requirement_met?` to clarify some of the expectations here.

Reply:
I think a rename makes sense. Especially since this returns `true` or `false` and doesn't have side-effects. I think a predicate method name like `selfie_require_met?` works.